dbt/include/athena/macros/materializations/models/incremental/helpers.sql

```diff
@@ -54,3 +57,37 @@
     {%- do adapter.clean_up_partitions(target_relation.schema, target_relation.table, partitions[i]) -%}
   {%- endfor -%}
 {%- endmacro %}
+
+{% macro remove_partitions_from_columns(columns_with_partitions, partition_keys) %}
+  {%- set columns = [] -%}
+  {%- for column in columns_with_partitions -%}
+    {%- if column.name not in partition_keys -%}
+      {%- do columns.append(column) -%}
+    {%- endif -%}
+  {%- endfor -%}
+  {{ return(columns) }}
+{%- endmacro %}
+
+{% macro ddl_data_type(col_type) -%}
+    -- transform varchar
+  {% set re = modules.re %}
+  {% set ns = namespace(ddl_type=col_type) %}
```

Comment:
We need to be careful, we have two identical implementations of the same thing, and we just changed it yesterday on the iceberg helpers, we might want to factorize this a little.

Reply:
if the implementation is identical, let's move it under  models/utils/ddl_data_type.sql
and then use that in iceberg too.

Reply:
I think so.
Can I move this to that place and commonoze it?

Reply:
Absolutely, that was the suggestion @Jrmyy , and picking up from there I suggest you to:
* move the macro into: `models/utils/ddl_data_type.sql`
* refactor iceberg to use that new macro
* use the new macro on this on_schema change

Reply:
We could add another little section to replace character with an empty value, and then let ddl type handling do the rest as now.